cabal-install/cabal-install.cabal

```diff
@@ -20,60 +17,38 @@ Copyright:          2003-2020, Cabal Development Team
 Category:           Distribution
 Build-type:         Simple
 Extra-Source-Files:
-  README.md bash-completion/cabal changelog
+  README.md
+  bash-completion/cabal
+  changelog
 
 source-repository head
   type:     git
   location: https://github.com/haskell/cabal/
   subdir:   cabal-install
 
 Flag native-dns
-  description:  Enable use of the [resolv](https://hackage.haskell.org/package/resolv) & [windns](https://hackage.haskell.org/package/windns) packages for performing DNS lookups
+  description:
+    Enable use of the [resolv](https://hackage.haskell.org/package/resolv)
+    & [windns](https://hackage.haskell.org/package/windns) packages for performing DNS lookups
   default:      True
   manual:       True
 
-Flag debug-expensive-assertions
-  description:  Enable expensive assertions for testing or debugging
-  default:      False
-  manual:       True
-
-Flag debug-conflict-sets
-  description:  Add additional information to ConflictSets
-  default:      False
-  manual:       True
-
-Flag debug-tracetree
-  description:  Compile in support for tracetree (used to debug the solver)
-  default:      False
-  manual:       True
-
 Flag lukko
   description:  Use @lukko@ for file-locking
   default:      True
   manual:       True
 
-executable cabal
-    main-is:        Main.hs
-    hs-source-dirs: main
-    default-language: Haskell2010
-    ghc-options:    -Wall -fwarn-tabs -fwarn-incomplete-uni-patterns -fwarn-incomplete-record-updates
-    if impl(ghc >= 8.0)
-        ghc-options: -Wcompat
-                     -Wnoncanonical-monad-instances
-      if impl(ghc < 8.8)
-        ghc-options: -Wnoncanonical-monadfail-instances
 
+library
```

Comment:
There are relevant issues, like
- https://github.com/haskell/cabal/issues/6090
- https://github.com/haskell/cabal/issues/7224
- https://github.com/haskell/cabal/issues/3781

Is this trying to address them?

Reply:
This is actually in service of coverage reporting, but if it fixes them, great.